lib/tokenFactory.js

```diff
@@ -0,0 +1,116 @@
+}
+
+/**
+ * Unseal token value and return the token with the unsealed value
+ * @param  {TokenModel}         token   Token to unseal
+ * @param  {String}             pw      Password to unseal with[description]
+ * @return {Promise}
+ */
+function unsealToken(token, pw) {
+    return nodeify.withContext(iron, 'unseal', [token.value, pw, iron.defaults])
+        .then((unsealed) => {
+            token.value = unsealed;
+
+            return token;
+        });
+}
+
+class TokenFactory extends BaseFactory {
+    /**
+     * Construct a TokenFactory object
+     * @method constructor
+     * @param  {Object}    config
+     * @param  {Object}    config.datastore     Object that will perform operations on the datastore
+     * @param  {String}    config.password      Password for encryption operations
+     */
+    constructor(config) {
+        super('token', config);
+        this[password] = config.password;
+    }
+
+    /**
+     * Instantiate a Token class
+     * @method createClass
+     * @param  config
+     * @return {Token}
+     */
+    createClass(config) {
+        const c = config;
+
+        c.password = this[password];
+
+        return new Token(c);
+    }
+
+    /**
+     * Create a token model
+     * Need to seal the token before saving
+     * @method create
+     * @param  {Object}     config
+     * @param  {String}     config.userId            The ID of the associated user
+     * @param  {String}     config.value             The token value
+     * @param  {String}     config.name              The token name
+     * @param  {String}     config.description       The token description
+     * @return {Promise}
+     */
+    create(config) {
+        return sealToken(config.value, this[password])
+            .then((sealed) => {
+                config.value = sealed;
+                config.lastUsed = null;
+
+                return super.create(config);
+            });
+    }
+
+    /**
+     * List tokens with pagination and filter options
+     * @method list
+     * @param  {Object}   config                  Config object
+     * @param  {Object}   config.params           Parameters to filter on
+     * @param  {Object}   config.paginate         Pagination parameters
+     * @param  {Number}   config.paginate.count   Number of items per page
+     * @param  {Number}   config.paginate.page    Specific page of the set to return
+     * @return {Promise}
+     */
+    list(config) {
+        return super.list(config).then(tokens =>
+            Promise.all(tokens.map(token => unsealToken(token, this[password]))));
+    }
+
+    /**
+     * Get an instance of the TokenFactory
+     * @method getInstance
+     * @param  {Object}     config
+     * @param  {Datastore}  config.datastore    A datastore instance
+     * @param  {String}     config.password     Password for encryption operations
+     * @return {TokenFactory}
+     */
+    static getInstance(config) {
```

Comment:
I had thought that was important from looking at other factories, but having gone back and looked at where the other factories getInstance methods are called I don't think it's actually necessary for tokens. It seems to be for when one object with a factory has another object with a factory as an attribute, and as far as I know there isn't going to be anything that uses a token like that.